nuitka/importing/Importing.py

```diff
@@ -141,12 +141,13 @@ def getModuleNameAndKindFromFilename(module_filename):
     """
 
     # TODO: This does not handle ".pyw" files it seems.
-    if os.path.isdir(module_filename):
-        module_name = ModuleName(os.path.basename(module_filename))
-        module_kind = "py"
-    elif module_filename.endswith(".py"):
+    if module_filename.endswith(".py"):
         module_name = ModuleName(os.path.basename(module_filename)[:-3])
         module_kind = "py"
+
+    elif os.path.isdir(module_filename):
+        module_name = ModuleName(os.path.basename(module_filename))
+        module_kind = "py"
```

Comment:
Maybe if the branches became "return driven", there wouldn't have to be elif sequences, and the final `else` would not impose a limit on where you can easily move this branch. I will just do that myself I guess.